CANTroller2/src/cantroller2.cpp

```diff
@@ -202,39 +206,33 @@ void loop() {
     // printf ("it:%d ac:%ld lst:%d ta:%d sc:%d el:%ld\n", boot_button, boot_button_action, boot_button_last, boot_button_timer_active, boot_button_suppress_click, dispResetButtonTimer.elapsed());
     
     // External digital signals - takes 11 us to read
-    if (!(simulating && sim_basicsw)) basicmodesw = !digitalRead (basicmodesw_pin);   // 1-value because electrical signal is active low
-    // if (ctrl == JOY && (!simulating || !sim_cruisesw)) cruise_sw = digitalRead (joy_cruise_btn_pin);
+    if (!simulator.simulating(SimOption::basicsw)) basicmodesw = !digitalRead (basicmodesw_pin);   // 1-value because electrical signal is active low
+    // if (ctrl == JOY && (!simulator.get_enabled() || !sim_cruisesw)) cruise_sw = digitalRead (joy_cruise_btn_pin);
 
     if (timestamp_loop) loop_savetime (looptimes_us, loopindex, loop_names, loop_dirty, "pre");
 
     if (take_temperatures) temp_soren();
-    if (sim_coolant && pot_overload == coolant) temps_f[ENGINE] = map (pot_filt_percent, 0.0, 100.0, temp_sensor_min_f, temp_sensor_max_f);
+    if (simulator.can_simulate(SimOption::coolant) && simulator.get_pot_overload() == SimOption::coolant) temps_f[ENGINE] = map (pot.get_filtered_value(), 0.0, 100.0, temp_sensor_min_f, temp_sensor_max_f);
```

Comment:
Comments or something, it's not obvious to me what this is doing. If the simulator can simulate coolant, and the value of the potentiometer overload is the same as the value of the coolant, set the engine temp equal to the output of a custom map function. The map function could also use to comments explaining what it's doing.

Reply:
agreed, its not very clear, tho i don't think this change is any worse than what we had before. And actually all this will disappear into classes as we convert these sensors to devices (like was done for PressureSensor), so we might not need to worry about this too much. I guess I could make a function is_pot_overload(SimOption::coolant) or something if you think that reads better. the map is just taking the pot value (pot_filt_percent) and mapping it between 0-100, its used all over (not that it couldn't use comments as well...)